File: discord_bot/slash_commands/hello_world.py
import discord
from discord import app_commands


def add_hello_world_slash_command(bot: discord.Client, guild: discord.Object):
    @bot.tree.command(name="hello-world", description="Showcase of a basic slash command", guild=guild)
    async def hello_world(interaction: discord.Interaction):
        await interaction.response.send_message("Hello world!")

File: go/discord_bot/slash_commands/hello_world.go
package slash_commands

import (
	"github.com/bwmarrin/discordgo"
)

var HelloWorldCommand = &discordgo.ApplicationCommand{
	Name:        "hello-world",
	Description: "Showcase of a basic slash command",
}

func HandleHelloWorld(s *discordgo.Session, i *discordgo.InteractionCreate) {
	s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
		Type: discordgo.InteractionResponseChannelMessageWithSource,
		Data: &discordgo.InteractionResponseData{
			Content: "Hello world!",
		},
	})
}
